subxt/src/blocks/extrinsic_types.rs

```diff
@@ -707,12 +726,24 @@ impl<'a> ExtrinsicSignedExtension<'a> {
         self.ty_id
     }
 
+    /// DecodedValueThunk representing the type of the extra of this signed extension.
+    fn decoded(&self) -> Result<DecodedValueThunk, Error> {
+        let decoded_value_thunk = DecodedValueThunk::decode_with_metadata(
+            &mut &self.bytes[..],
+            self.ty_id,
+            &self.metadata,
+        )?;
+        Ok(decoded_value_thunk)
+    }
```

Comment:
It looks like some things, like this and the lifetime stuff on `ExtrinsicSignedExtensions` were lost again compared to the base PR that these build off; would be good to merge the underlying branch or rebase :)

Reply:
You are right, some stuff was lost in the merge, sorry. But this `decoded` function I kept on purpose. It is just private now, but used in the public `as_type()` and `value()` functions.

Reply:
In the other PR I'd updated to directly use `Value::decode_as_type(..)` which should be better than `DecodedValueThunk::decode_with_metadata(..)`, and also I'd remove the  `value()` function and just return a `Value` from this one (I'd have a look at the other PR because that was good here :))